hack/errata-check.sh

```diff
@@ -0,0 +1,262 @@
+# diff -u <(git rev-list --first-parent $NEXT_RELEASE) \
+#         <(git rev-list --first-parent $LAST_RELEASE) \
+#         | sed -ne 's/^ //p' | head -1
+
+# Set variables
+ERRATA_API_URL="https://errata.devel.redhat.com/api/v1/erratum"
+RED=$(tput setaf 1)
+GREEN=$(tput setaf 2)
+YELLOW=$(tput setaf 3)
+NORMAL=$(tput sgr0)
+BLUE=$(tput setaf 4)
+BRIGHT=$(tput bold)
+
+# Define functions
+function help() {
+  printf "${YELLOW}"
+  echo "Usage: errata-check.sh branch_name"
+  echo "List merge commits in the build attached to an active release errata"
+  echo ""
+  echo "Example:"
+  echo "# Check the build for the next WMCO v7 errata"
+  echo "errata-check.sh release-4.12"
+  printf "${NORMAL}"
+}
+
+function verify-args {
+  if [ $# -eq 0 ]
+  then
+    print-error "No branch provided"
+    help
+    exit 1
+  fi
+}
+
+function print-label {
+  local label=$1
+  local item=$2
+  printf "${BLUE}%-20s= ${NORMAL}%s\n" "$label" "$item"
+}
+
+function print-info {
+  local text=$1
+  printf "\n${BRIGHT}%s${NORMAL}\n" "$text"
+}
+
+function print-error {
+  local text=$1
+    printf "\n${RED}ERROR: %s${NORMAL}\n" "$text" 1>&2
+}
+
+function set-branch-vars {
+  local branch=$1
+  case $branch in
+    release-4.14)
+      WMCO_VERSION=9
+      LAST_WMCO_VERSION="9.0.0 Branching Day"
+      LAST_WMCO_HASH="83eb755a1b225d40852040ef7deadd28ed93c769"
+      ;;
+    release-4.13)
+      WMCO_VERSION=8
+      LAST_WMCO_VERSION="8.0.0 Branching Day"
+      LAST_WMCO_HASH="b652b5c1918b6a1be827a1097269f9681022e077"
+      ;;
+    release-4.12)
+      WMCO_VERSION=7
+      LAST_WMCO_VERSION="v7.0.1"
+      LAST_WMCO_HASH="bc9473bf8c4200b0a016608e808122ef783570bc"
+      ;;
+    release-4.11)
+      WMCO_VERSION=6
+      LAST_WMCO_VERSION="v6.0.0"
+      LAST_WMCO_HASH="f3ad235ac91d573b1d64eed280618fd04ea8ce5f"
+      ;;
+    release-4.10)
+      WMCO_VERSION=5
+      LAST_WMCO_VERSION="v5.1.0"
+      LAST_WMCO_HASH="b49f4519f2be17b47d54397f87dffad5d7f077cc"
+      ;;
+    release-4.9)
+      WMCO_VERSION=4
+      LAST_WMCO_VERSION="v4.0.1"
+      LAST_WMCO_HASH="f66f0980845d450a821b23f7fc3a53cc82ccb7d6"
+      ;;
+    *)
+      print-error "Branch $branch is unsupported"
+      exit 1
+      ;;
+  esac
+  print-label "WMCO Version" "v$WMCO_VERSION"
+}
+
+function check-upstream-status {
+  # Checks if the local repo is up to date with
+  local branch=$1
+  print-label "Branch" "$branch"
+  local local_head
+  local upstream_head
+  local_head=$(git rev-parse "$branch")
+  upstream_head=$(git ls-remote upstream -h "refs/heads/$branch" | awk '{ print $1 }')
+  if [ "$local_head" != "$upstream_head" ]; then
+    print-error "Local branch $branch is not up to date with upstream"
+    exit 1
+  fi
+}
+
+function curl-json {
+  local url=$1
+  local result
+  if ! result=$(curl --silent --negotiate --user : "$url"); then
+    print-error "Unable to curl server.  Connected to VPN?"
+    exit 1
+  fi
+  if echo -E "$result" | grep "Errata Tool - Unauthorized"; then
```

Comment:
Nit: I suggest just checking for `Unauthorized` given `curl-json` is generic and could be used to curl URLs other than the errata website.

Reply:
Updated.  I now check the result code of curl as well as HTTP headers for errors before pulling content.